python/Gaffer/__init__.py

```diff
@@ -61,3 +64,19 @@
 
 # Class-level non-UI metadata registration
 Metadata.registerValue( Reference, "childNodesAreReadOnly", True )
+
+def rootPath() :
+
+	return pathlib.Path( os.path.expandvars( "$GAFFER_ROOT" ) )
+
+# Returns the path of the Gaffer executable for the current platform.
+# If `resolved` is `True`, the full path will be returned, otherwise
+# only the executable name and extension are returned.
+def executablePath( resolve = False ) :
+
+	executable = "gaffer" if os.name != "nt" else "gaffer.cmd"
+
+	if resolve :
+		return str( rootPath() / "bin" / executable )
```

Comment:
I had assumed we'd return a `pathlib.Path()`, since the method is called `executablePath()`. I checked, and it seems it would be OK to also return a `Path` when `resolve == False`. Is there anything I'm missing?

Reply:
Windows can't take a `pathlib.Path` directly in the argument list - it errors saying that it's not iterable. I think because it's going through the string and looking for things like quotes so it can format them for the command.

So I think we need to do `str( path )` either at the `subprocess.*` site or here. I figured this is less error prone, but up for other ideas.

Reply:
[This](https://bugs.python.org/issue31961) seems to imply that this is fixed in Python 3.8. If that's the case, it seems a pity to tie ourselves to strings...although I agree that putting `str()` everywhere for a while will be a pain...I suppose we could always change the return type later...what do you think?

Reply:
That's a nice find! Ugh, it doesn't feel right returning `str` knowing we'll change the API soon. I'll do a very quick squash in of the few places we use it, converting to `str` at the usage site. Last last-minute fix!

Reply:
That's done in 984f0088272e66f2a178b4cb62f7c92998d7db00